File: node-js/match_name.js
const OS_API_KEY = process.env.OS_API_KEY;
if (!OS_API_KEY) {
  console.log('Please set the OS_API_KEY environment variable');
  process.exit(1);
}

async function match() {
  const headers = {
    // Prepare authentication using the Authorization header
    "Authorization": OS_API_KEY,

    // We're going to encode the request body as JSON.
    'Content-Type': 'application/json',
  };

  // Prepare a query to match on schema and the name property
  const requestBody = {
    "queries": {
      "q1": {
        "schema": "Person",
        "properties": {
          "name": ["Barack Obama"]
        }
      }
    }
  }

  // Make the request
  const response = await fetch(
    'https://api.opensanctions.org/match/default',
    {
      'method': 'POST',
      'headers': headers,
      'body': JSON.stringify(requestBody),
    });

  // Check for errors
  if (!response.ok) {
    console.error(await response.text());
    throw new Error('HTTP error\n' + response.status + ' ' + response.statusText);
  }

  // Get the results for our query
  const responseBody = await response.json();
  // We're just JSON encoding the response data here for pretty printing
  console.log(JSON.stringify(responseBody.responses.q1.results, null, 2));
}

match()
  .then()
  .catch((error) => console.log("Error:" + error.message));

File: python/match_name.py
from pprint import pprint
import requests
import os

OS_API_KEY = os.getenv("OS_API_KEY")
if not OS_API_KEY:
    raise ValueError("The OS_API_KEY environment variable is not set")


def main():
    # Prepare a query to match on schema and the name property
    query = {
        "queries": {
            "q1": {"schema": "Person", "properties": {"name": ["Barack Obama"]}}
        }
    }
    # Authenticate using the API key in the Authorization header
    headers = {
        "Authorization": OS_API_KEY,
    }
    response = requests.post(
        "https://api.opensanctions.org/match/default", headers=headers, json=query
    )
    # Check for HTTP errors
    response.raise_for_status()

    # Get the results for our query
    q1_results = response.json()["responses"]["q1"]["results"]

    # print it out with nice formatting
    pprint(q1_results, sort_dicts=False)


if __name__ == "__main__":
    main()
